client/my-sites/site-settings/jetpack-credentials/credentials-form/index.jsx

```diff
@@ -41,8 +41,17 @@ export class CredentialsForm extends Component {
 	};
 
 	handleFieldChange = event => {
+		const newForm = {
+			...this.state.form,
+			[ event.target.name ]: event.target.value,
+		};
+
+		if ( 'protocol' === event.target.name ) {
+			newForm.port = 'ftp' === event.target.value ? 21 : 22;
+		}
+
 		this.setState( {
-			form: { ...this.state.form, [ event.target.name ]: event.target.value },
+			form: newForm,
```

Comment:
nothing beats `Object.assign()` for conditional properties…

```js
handleFieldChange = ( { target: { name, value } } ) => {

const form = Object.assign(
	'protocol' === name && { port: 'ftp' === value ? 21 : 22 },
	this.state.form,
	{ [ name ]: value },
);
```

Reply:
```js
const changedProtocol = 'protocol' === name;
const port = 'ftp' === value ? 21 : 22;

const form = Object.assign(
	changedProtocol && { port },
	this.state.form,
	{ [ name ]: value },
);
```

Reply:
Neatened this up a bit in https://github.com/Automattic/wp-calypso/pull/19338/commits/825227d13ea52178bbd9641d8357882e30667eff - can you take another look? Thanks!